### ISSUE TITLE
The question about the selection of semantic segmentation network

### ISSUE THREAD

**Issue by super233** [lorenmt/reco#12]
Hi, thanks for your awesome work.

I have a question about the selection of semantic segmentation networks in contrastive learning, which is not about the code.

I have noticed that recent contrastive learning based works are all implemented with DeepLab V3+, such as [DCL](http://arxiv.org/abs/2106.14133), [PC2Seg](http://arxiv.org/abs/2108.09025), [RegionContrast](https://openaccess.thecvf.com/content/ICCV2021/html/Hu_Region-Aware_Contrastive_Learning_for_Semantic_Segmentation_ICCV_2021_paper.html) and your paper, and I am curious about why there's no paper to implement contrastive learning on DeepLab V2.

In my view, it's mainly due to the shallow structure of DeepLab V2, which is not convenient to generate well representational features for subsequent contrastive learning. In your implementation of DeepLab V2, the features are generated from the backbone, however, the predicted logits are generated from the ASPP.

What do you think about this question? Looking forward to your reply. Please forgive me for interrupting. :-)

**Comment by lorenmt** [comment]
The design of DeepLabv2 directly uses ASPP module (based on encoder feature) for label prediction which is not well suited to attach an additional branch for representation learning.

In addition, DeepLabv2 trains significantly slower than DeepLabv3+ with worse performance. So choosing DeepLabv3+ seems to be a no-brainer.
